tecken/base/symbolstorage.py

```diff
@@ -132,40 +125,66 @@ def _get(self, symbol, debugid, filename):
         was returned as an indication that the symbol actually exists.
 
         """
-        for i, source in enumerate(self.sources):
-            prefix = source.prefix
+        if try_storage:
+            assert self.try_backend is not None
+            backends = [*self.backends, self.try_backend]
+        else:
+            backends = self.backends
+        for backend in backends:
+            prefix = backend.prefix
             assert prefix
 
             # We'll put together the URL manually
             file_url = "{}/{}".format(
-                source.base_url, self.make_url_path(prefix, symbol, debugid, filename)
+                backend.base_url, self.make_url_path(prefix, symbol, debugid, filename)
             )
             logger.debug(f"Looking for symbol file by URL {file_url!r}")
             if last_modified := get_last_modified(file_url):
                 age_days = int(time.time() - last_modified) // 86_400  # seconds per day
-                if i == self.try_url_index:
+                if backend.try_symbols:
                     tags = ["storage:try"]
                 else:
                     tags = ["storage:regular"]
                 METRICS.histogram("symboldownloader.file_age_days", age_days, tags)
                 return {
                     "url": file_url,
-                    "source": source,
+                    "backend": backend,
                 }
 
     @set_time_took
-    def has_symbol(self, symbol, debugid, filename):
+    def has_symbol(
+        self, symbol: str, debugid: str, filename: str, try_storage: bool = False
+    ) -> bool:
         """return True if the symbol can be found, False if not
         found in any of the URLs provided."""
-        return bool(self._get(symbol, debugid, filename))
+        return bool(self._get(symbol, debugid, filename, try_storage))
 
     @set_time_took
-    def get_symbol_url(self, symbol, debugid, filename):
+    def get_symbol_url(
+        self, symbol: str, debugid: str, filename: str, try_storage: bool = False
+    ) -> str:
         """Return the redirect URL or None.
 
         If we return None it means we can't find the object in any of the URLs provided.
 
         """
-        found = self._get(symbol, debugid, filename)
+        found = self._get(symbol, debugid, filename, try_storage)
         if found:
             return found["url"]
+
+
+# Global SymbolStorage instance, eventually used for all interactions with storage backends.
+_symbol_storage = None
+
+
+def symbol_storage(force_recreate: bool = False) -> SymbolStorage:
```

Comment:
I would use the appconfig model because it makes sure things happen during webapp bootstrapping. If the code evolved such that it was more intertwined with things, having it get instantiated in a predictable time helps a lot.